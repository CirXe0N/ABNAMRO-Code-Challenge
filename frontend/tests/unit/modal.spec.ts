import { shallowMount } from '@vue/test-utils';
import ModalComponent from '@/components/Modal.vue';

describe('Modal.vue', () => {
  it('renders props.node when passed', () => {
    const node = { name: 'Test Node', description: 'Test Description' };
    const wrapper = shallowMount(ModalComponent, {
      propsData: { node },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('close the modal', () => {
    const node = { name: 'Test Node', description: 'Test Description' };
    const wrapper = shallowMount(ModalComponent, {
      propsData: { node },
    });

    (wrapper.vm as any).close();

    expect((wrapper.vm as any).isOpen).toBeFalsy();
  });

  it('open the modal', () => {
    const node = { name: 'Test Node', description: 'Test Description' };
    const wrapper = shallowMount(ModalComponent, {
      propsData: { node },
    });

    (wrapper.vm as any).open();

    expect((wrapper.vm as any).isOpen).toBeTruthy();
  });
});
