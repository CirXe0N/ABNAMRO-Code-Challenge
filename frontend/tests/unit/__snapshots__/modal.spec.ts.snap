// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal.vue renders props.node when passed 1`] = `
<div
  class="dialog"
>
  <div
    class="dialog__overlay"
  />
   
  <div
    class="dialog__content"
  >
    <div
      class="morph-shape"
    >
      <svg
        height="100%"
        preserveAspectRatio="none"
        viewBox="0 0 560 280"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="276"
          width="556"
          x="3"
          y="3"
        />
      </svg>
    </div>
     
    <div
      class="dialog-inner"
    >
      <h2>
        Node: Test Node
      </h2>
       
      <p>
        Test Description
      </p>
       
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;
